src/datatype/memory_bvecf32.rs

```diff
@@ -152,6 +153,31 @@ impl IntoDatum for BVecf32Output {
     }
 }
 
+impl FromDatum for BVecf32Output {
+    unsafe fn from_polymorphic_datum(datum: Datum, is_null: bool, _typoid: Oid) -> Option<Self> {
+        if is_null {
+            None
+        } else {
+            let ptr = NonNull::new(datum.cast_mut_ptr::<BVecf32Header>()).unwrap();
+            let q = unsafe {
```

Comment:
It's unsound: if it's not detoasted, the pointer will be freed by `drop_in_place::<BVecf32Header>`, and it may lead to use-after-free.